profile/special.go

```diff
@@ -117,3 +138,63 @@ func getSpecialProfile(profileID, linkedPath string) *Profile {
 		return nil
 	}
 }
+
+// specialProfileNeedsReset is used as a workaround until we can properly use
+// profile layering in a way that it is also correctly handled by the UI. We
+// check if the special profile has not been changed by the user and if not,
+// check if the profile is outdated and can be upgraded.
+func specialProfileNeedsReset(profile *Profile) bool {
+	switch {
+	case profile.Source != SourceLocal:
+		// Special profiles live in the local scope only.
+		return false
+	case profile.LastEdited > 0:
+		// Profile was edited - don't override user settings.
+		return false
+	}
```

Comment:
Should we create a notification for that case so the user knows that we updated the defaults?